.github/workflows/sync-cac-oscal.yml

```
@@ -0,0 +1,413 @@
+              echo "CHANGE_FOUND=true" >> $GITHUB_ENV
+              break
+            fi
+          done
+          cat filenames.txt
+        env:
+          GH_TOKEN: ${{ env.INSTALLATION_TOKEN }}
+      # Step 6: Setup the trestle bot environment
+      - name: Checkout OSCAL content repo
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        uses: actions/checkout@v4
+        with:
+          repository: ComplianceAsCode/oscal-content
+          path: oscal-content
+          token: ${{ env.INSTALLATION_TOKEN }}
+          fetch-depth: 0
+      - name: Checkout trestle-bot repo
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        uses: actions/checkout@v4
+        with:
+          repository: complytime/trestle-bot
+          path: trestle-bot
+      - name: Setup trestlebot
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        run: |
+          cd trestle-bot && python3 -m venv venv && source venv/bin/activate
+          python3 -m pip install --no-cache-dir "poetry==1.7.1"
+          poetry install
+      # Step 7: Get profiles, controls and level mapping
+      - name: Get profiles, controls and level mapping
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        run: |
+          cd trestle-bot && source venv/bin/activate
+          RHEL_PRODUCTS=(rhel8 rhel9 rhel10)
+          for product in "${RHEL_PRODUCTS[@]}"; do
+            echo "The map for $product..."
+            map_file=$product"_map.json"
+            python scripts/get_mappings_profile_control_levels.py $product "$GITHUB_WORKSPACE/cac-content" > $map_file  2>&1
+            cat $map_file
+          done
+      # Step 8: Get product available controls
+      - name: Get product controls
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        run: |
+          cd trestle-bot && source venv/bin/activate
+          # RHEL_PRODUCTS=(rhel8 rhel9 rhel10 ocp4 rhv4 rhcos4)
+          RHEL_PRODUCTS=(rhel8 rhel9 rhel10)
+          for product in "${RHEL_PRODUCTS[@]}"; do
+            echo "All available controls of $product..."
+            controls_file=$product"_controls"
+            python scripts/get_product_controls.py $product "$GITHUB_WORKSPACE/cac-content" > $controls_file  2>&1
+            cat $controls_file
+          done
+      # Step 9: Handle the detected updates
+      # 1. Get the updated controls
+      # 2. Get the updated profiles
+      # 3. Get the controls and profiles are impacted by rules and vars
+      - name: Handle the detected updates
+        if: ${{ env.CHANGE_FOUND == 'true' }}
+        run: |
+          cat filenames.txt
+          cd trestle-bot && source venv/bin/activate
+          RHEL_PRODUCTS=(rhel8 rhel9 rhel10)
+          while IFS= read -r line; do
+            # 1. Get the updated controls
+            if [[ "$line" == *controls/* ]]; then
+              echo "Detect update for control $line ......"
+              control=$(echo $line | rev | cut -f1 -d"/" | rev | cut -f1 -d".")
```

```suggestion
              control=$(basename $line | cut -f1 -d".")
```
This is simpler and should have the same result. Could you update on other steps as well, please?